nipoppy/workflows/dataset_init.py

```diff
@@ -57,20 +60,41 @@ def run_main(self):
         Copy boutiques descriptors and invocations.
         Copy default config files.
 
+
         If the BIDS source dataset is requested, it is copied.
+        If dataladd is used, it is installed with datalad.
         """
         # dataset must not already exist
         if self.dpath_root.exists():
             raise FileExistsError("Dataset directory already exists")
 
+        self._init_as_datalad_dataset()
+
         # create directories
         for dpath in self.layout.dpaths:
 
-            # If a bids_source is passed it means datalad is installed.
-            if self.bids_source is not None and dpath.stem == "bids":
-                self.copytree(self.bids_source, str(dpath), log_level=logging.DEBUG)
-            else:
+            if dpath.stem != "bids" or self.bids_source is None:
                 self.mkdir(dpath)
+                continue
+
+            if not self.use_dalatad:
+                self.copytree(self.bids_source, str(dpath), log_level=logging.DEBUG)
+                continue
+
+            from datalad import api
+
+            self.logger.info(
+                f"Installing datalad BIDS raw dataset from {self.bids_source}."
+            )
+            dataset = None
+            dataset = self.dpath_root
+
+            api.install(
+                dataset=dataset,
+                path=dpath,
```

Comment:
note that if the source bids dataset is not a datalad dataset, then it would be better to copy it and then save it as a datalad dataset